server/routes/checkAnswersController.test.ts

```diff
@@ -29,89 +42,118 @@ beforeEach(() => {
   expectedArrivalsService.createOffenderRecordAndBooking.mockResolvedValue({
     offenderNo: 'A1234AB',
   })
+  imprisonmentStatusesService.getReasonForImprisonment.mockResolvedValue(
+    'Determinate sentence - Extended sentence for public protection'
+  )
 })
 
 afterEach(() => {
   jest.resetAllMocks()
 })
 
-describe('GET /checkAnswers', () => {
-  it('should redirect to authentication error page for non reception users', () => {
-    app = appWithAllRoutes({ roles: [] })
-    return request(app).get('/prisoners/12345-67890/check-answers').expect(302).expect('Location', '/autherror')
-  })
+const hourFromNow = moment().add(1, 'hour').format('ddd, D MMM yyyy HH:mm:ss [GMT]').toString()
 
-  it('should call service method correctly', () => {
-    return request(app)
-      .get('/prisoners/12345-67890/check-answers')
-      .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect(res => {
-        expect(expectedArrivalsService.getMove).toHaveBeenCalledTimes(1)
-        expect(expectedArrivalsService.getMove).toHaveBeenCalledWith('12345-67890')
-      })
-  })
+describe('/checkAnswers', () => {
+  describe('view()', () => {
+    it('should redirect to authentication error page for non reception users', () => {
+      app = appWithAllRoutes({ roles: [] })
+      return request(app).get('/prisoners/12345-67890/check-answers').expect(302).expect('Location', '/autherror')
+    })
 
-  it('should render /check-answers page', () => {
-    return request(app)
-      .get('/prisoners/12345-67890/check-answers')
-      .expect('Content-Type', 'text/html; charset=utf-8')
-      .expect(res => {
-        const $ = cheerio.load(res.text)
-        expect($('h1').text()).toContain('Check your answers before adding')
-      })
+    it('should get status and reason from cookie and call service methods correctly', () => {
+      return request(app)
+        .get('/prisoners/12345-67890/check-answers')
+        .set(
+          'cookie',
+          `status-and-reason=s%3Aj%3A%7B%22code%22%3A%22determinate-sentence%22%2C%22imprisonmentStatus%22%3A%22SENT%22%2C%22movementReasonCode%22%3A%2226%22%7D.QEx%2B2EcyCfMkSknBJwkaVswIBLsUTbGFLkXur2qN%2Fro; Max-Age=7200; Domain=localhost; Path=/; Expires=${hourFromNow}; HttpOnly; SameSite=Lax`
+        )
+        .expect('Content-Type', 'text/html; charset=utf-8')
+        .expect(res => {
+          expect(expectedArrivalsService.getMove).toHaveBeenCalledTimes(1)
+          expect(expectedArrivalsService.getMove).toHaveBeenCalledWith('12345-67890')
```

Comment:
meh.. future reference - don't need to do both

```
          expect(expectedArrivalsService.getMove).toHaveBeenCalledTimes(1)
          expect(expectedArrivalsService.getMove).toHaveBeenCalledWith('12345-67890')
```

just the calledWith version is enough, its probably over specifying the behaviour saying how many times its being called. calledWith will fail if not called at all

Reply:
Cool, I'll remove the toHaveBeenCalledTimes as a tidy up in the next PR